docs/src/architecture/assets.md

```diff
@@ -0,0 +1,57 @@
+
+* Asset exchange should be parallelizable 
+* Asset ownership should be private 
+* Asset usage should be truly censorship resistant
+* Fees can be paid using any asset
+
+## Definition
+Assets are data structures that can be stored in account and note vaults. In Miden, there are fungible and non-fungible assets. All assets are encoded using a single `Word` (4 elements) such that it is easy to determine the type of an asset both inside and outside Miden VM. 
+
+Specifically, the first element will be
+
+   * `ZERO` for a fungible asset: `[e0, ZERO, e2, e3]`
+   * `non-ZERO` for a non-fungible asset: `[e0, non-ZERO, e2, e3]`
+  
+and the most significant bit will be
+
+   * `ONE` for a fungible asset
+   * `ZERO` for a non-fungible asset
+
+The above properties guarantee that there can never be a collision between a fungible and a non-fungible asset. 
+
+## Issuance
+Specialized asset accounts called faucets issue assets by creating notes. Like any other account faucets have an Account ID called `faucet_id`. Users will be able to create faucets on their own. This is similar to creating ERC20 contracts on Ethereum. The methodology for constructing fungible and non-fungible assets is described below.
+
+### Fungible assets
+A fungible asset consists of a `faucet_id` of the faucet which issued the asset as well as the asset amount. The asset amount is guaranteed to be `2^63 - 1` or smaller.
+
+The most significant element of a fungible asset is set to the ID of the faucet which issued
+the asset. This guarantees the properties described above (the first bit is `ONE`).
+
+The least significant element is set to the amount of the asset. This amount cannot be greater
+than `2^63 - 1` and thus requires 63-bits to store. Elements 1 and 2 are set to `ZERO`.
+
+Note: Miden uses little-endian, so a fungible asset looks like `[amount, ZERO, e2, faucet_id]` wheras `e3` as bit ends with `1`
+
+It is impossible to find a collision between two fungible assets issued by different faucets as
+the `faucet_id` is included in the description of the asset and this is guaranteed to be different
+for each faucet as per the faucet creation logic.
+
+### Non-fungible assets
+A non-fungible asset consists of 4 field elements which are computed by hashing asset data (which can be of arbitrary length) to produce: `[e0, e1, e2, e3]`.  We then replace `e1` with the `faucet_id` that issued the asset: `[e0, faucet_id, e2, e3]` to comply with the above rule. We then set the most significant bit to `ZERO`.
+
+Note: Miden uses little-endian, so a non-fungible asset looks like `[e0, faucet_id, e2, e3]` wheras `e3` as bit ends with `0`
+
+It is impossible to find a collision between two non-fungible assets issued by different faucets as the `faucet_id` is included in the description of the non-fungible asset and this is guaranteed to be different as per the faucet creation logic. Collision resistance for non-fungible assets issued by the same faucet is `~2^95`.
```

Comment:
I think most of the content here is way too detailed and should probably go into the specs section in the future. We should, however, highlight info that could be interesting to users/developers. For example:

- Max supply of any fungible asset is $2^{63} - 1$.
- Asset encodes both ID of the issuing account as well as asset details. So, when we look at an asset we know exactly who issued it.
- The rules for who can issue assets and how are fully customizable and defined by the code deployed in the faucet account. The only restrictions are: max supply for fungible assets and uniqueness of non-fungible assets.

It might also be good to give example of fungible and non-fungible assets (especially for latter). Like how something like a DevCon ticket (or something else) could be encoded as a non-fungible assets.

Regarding issuance, would be good to give examples too. Like when we issue an asset, it could be sent out immediately in a note. But it could also stay in the faucet account. This way, for example, we could mint a million NFTs off-chain in a single transaction, and then send them out as needed in separate transactions in the future.

There are probably other interesting tidbits like this that we can describe to give help the reader understand why certain decisions were made.

Reply:
I changed the section